types/errors.d.ts

```diff
@@ -67,6 +70,7 @@ declare namespace Errors {
   export class SocketError extends UndiciError {
     name: 'SocketError';
     code: 'UND_ERR_SOCKET';
+    socket: Socket | TLSSocket
```

Comment:
I think it's safer to have this typed as nullable.

Reply:
Agreed.